apollo-router/src/spec/query.rs

```diff
@@ -861,25 +840,27 @@ impl Query {
                     include_skip,
                     ..
                 } => {
-                    // top level objects will not provide a __typename field
-                    if type_condition.as_str() != root_type_name {
-                        return Err(InvalidValue);
-                    }
-
                     if include_skip.should_skip(parameters.variables) {
                         continue;
                     }
 
-                    self.apply_selection_set(
```

Comment:
We are still inside root selection, even if we go inside an inline fragment.
So it should be `apply_root_selection_set`

Reply:
this sounds reasonable but looks like a dangerous change. Do you have a test that would show what happens before and after that change?

Reply:
yes it tested here: https://github.com/apollographql/router/blob/e871c9331b9d1e7a66d646574f012f116f48b0c1/apollo-router/tests/integration/subgraph_response.rs#L34-L82